src/control/server/server_utils.go

```diff
@@ -89,6 +91,33 @@ func cfgGetRaftDir(cfg *config.Server) string {
 	return filepath.Join(cfg.Engines[0].Storage.Tiers.ScmConfigs()[0].Scm.MountPoint, "control_raft")
 }
 
+func writeCoreDumpFilter(log logging.Logger, path string, filter uint8) error {
+	// Add a workaround for a testing oddity that seems to be related to
+	// launching the server via SSH, with the result that the /proc file
+	// is owned by root and is therefore unwritable.
+	st, err := os.Stat(path)
+	if err != nil {
+		return errors.Wrapf(err, "unable to stat %s", path)
+	}
+	sys, ok := st.Sys().(*syscall.Stat_t)
+	if !ok {
+		return errors.New("os.Stat() did not return a *syscall.Stat_t")
+	}
+	if sys.Uid != uint32(syscall.Geteuid()) {
+		log.Debugf("%s is not owned by current user, skipping core dump filter", path)
+		return nil
+	}
```

Comment:
I wonder if it might be more efficient to attempt the open/write and check `os.IsPermission(err)` afterward, to get the same information

Reply:
Hmm. Yeah, I think you're right. I added all of this stuff while trying to figure out what the problem was, but there's not really a good reason to go through all of these steps in the normal case where it works OK.